components/ui-layout-view/src/UiLayoutView.types.ts

```diff
@@ -0,0 +1,48 @@
+import { ReactNode, Ref } from 'react';
+
+// ----- Sidebar definition
+
+type HasNoLeftSidebar = {
+  isLeftSidebarVisible?: never;
+  leftSidebar?: never;
+  leftSidebarClassName?: never;
+};
+
+type HasLeftSidebar = {
+  isLeftSidebarVisible: boolean;
+  leftSidebar: ReactNode;
+  leftSidebarClassName?: string;
+};
+
+type HasNoRightSidebar = {
+  isRightSidebarVisible?: never;
+  rightSidebar?: never;
+  rightSidebarClassName?: never;
+};
+
+type HasRightSidebar = {
+  isRightSidebarVisible: boolean;
+  rightSidebar: ReactNode;
+  rightSidebarClassName?: string;
+};
```

Comment:
How about to use renderer function, instead of `ReactNode` and additional flags?

This is better in my opinion by two reasons:

- types will be simpler, instead of flag and node, we expect a single renderer function;
- usage of renderer function is more performant, because we control when sidebar will be
  rendered.

Why I prefer renderer function. For example, we have a complex sidebar.

```tsx
const Sidebar: FC = () => (
  // some complex logic
);
```

And how we use it with ReactNode property.

```tsx
const Layout: FC = () => {
  const [isSidebarVisible] = useState(false);

  // ATTENTION: It runs all their hooks, logic and other every time.
  const sidebar = <Sidebar />;
  
  return (
    <UiLayoutView
      isRightSidebarVisible={isSidebarVisible}
      rightSidebar={sidebar}
    >
    // some content
    </UiLayoutView>
  );
}
```

Pay attention: we create a `sidebar` variable every render, even if `isSidebarVisible` is false. Or we should check this variable too, and render conditionally.

Instead, we can provide a renderer function, which will render node (especially, if it's complex) when sidebar is required.

If we render our property grid even when sidebar is closed, then we'll waste resources.

Actually, I think about this way to control render early, but in case of layout it's looking more useful, instead of small components, which can have a small piece of logic.

Reply:
I changed sidebar props to render fn, but

> types will be simpler, instead of flag and node, we expect a single renderer function;

We can't do this, because of flag is controlling visibility of sidebar.

Reply:
It's OK for me (=